### PR TITLE
set dynamic log levels configuration in yaml properties

### LINKED ISSUE #6709: [Feature] Set log levels configuration in YAML properties rather than logback.xml
##### **Overview of the issue**

Currently, log levels are setup in `src/main/resources/logback-spring.xml` as such:
```
    <logger name="io.github.jhipster.sample" level="#logback.loglevel#"/>
    <logger name="io.github.jhipster" level="#logback.loglevel#"/>

    <logger name="javax.activation" level="WARN"/>
    <logger name="javax.mail" level="WARN"/>
    <logger name="javax.xml.bind" level="WARN"/>
    <logger name="ch.qos.logback" level="WARN"/>
    <logger name="com.codahale.metrics" level="WARN"/>
    <logger name="com.ryantenney" level="WARN"/>
    <logger name="com.sun" level="WARN"/>
    <logger name="com.zaxxer" level="WARN"/>
    <logger name="io.undertow" level="WARN"/>
    <logger name="io.undertow.websockets.jsr" level="ERROR"/>
    <logger name="org.ehcache" level="WARN"/>
    <logger name="org.apache" level="WARN"/>
    <logger name="org.apache.catalina.startup.DigesterFactory" level="OFF"/>
    <logger name="org.bson" level="WARN"/>
    <logger name="org.hibernate.validator" level="WARN"/>
    <logger name="org.hibernate" level="WARN"/>
    <logger name="org.hibernate.ejb.HibernatePersistence" level="OFF"/>
    <logger name="org.springframework" level="WARN"/>
    <logger name="org.springframework.web" level="WARN"/>
    <logger name="org.springframework.security" level="WARN"/>
    <logger name="org.springframework.cache" level="WARN"/>
    <logger name="org.thymeleaf" level="WARN"/>
    <logger name="org.xnio" level="WARN"/>
    <logger name="springfox" level="WARN"/>
    <logger name="sun.rmi" level="WARN"/>
    <logger name="liquibase" level="WARN"/>
    <logger name="LiquibaseSchemaResolver" level="INFO"/>
    <logger name="sun.rmi.transport" level="WARN"/>

    <root level="#logback.loglevel#">
        <appender-ref ref="CONSOLE"/>
    </root>
```
However Spring Boot allows configuring those log levels as properties: 
```
logging:
    level:
        ROOT: #logback.loglevel#
        io.github.jhipster.sample:  #logback.loglevel#
        io.github.jhipster:  #logback.loglevel#
        javax.activation: WARN
        javax.mail: WARN
        javax.xml.bind: WARN
        ch.qos.logback: WARN
        com.codahale.metrics: WARN
        com.ryantenney: WARN
        com.sun: WARN
        com.zaxxer: WARN
        io.undertow: WARN
        io.undertow.websockets.jsr: ERROR
        org.ehcache: WARN
        org.apache: WARN
        org.apache.catalina.startup.DigesterFactory: OFF
        org.bson: WARN
        org.hibernate.validator: WARN
        org.hibernate: WARN
        org.hibernate.ejb.HibernatePersistence: OFF
        org.springframework: WARN
        org.springframework.web: WARN
        org.springframework.security: WARN
        org.springframework.cache: WARN
        org.thymeleaf: WARN
        org.xnio: WARN
        springfox: WARN
        sun.rmi: WARN
        liquibase: WARN
        LiquibaseSchemaResolver: INFO
        sun.rmi.transport: WARN
```
##### **Motivation for or Use Case**

- Handle the case of running the app without starting the maven build will cause the logback.xml file (YAML file treat `#` as comments)
- Set all configuration in one place
- For apps using the registry this is cleaner when defining those properties from the Registry (config server) as you know which property will take priority.

##### **Related issues**

https://github.com/jhipster/generator-jhipster/issues/6703

##### **JHipster Version(s)**

v4.10.2

##### **JHipster configuration**

All JHipster configurations will be affected

- [X] Checking this box is mandatory (this is just to show you read everything)

As requested by @jdubois this will be subject to a vote as it might break existing configurations. So please 👍 or 👎.

### PR DESCRIPTION
**Pull request by PierreBesson**
fix #6709

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/main/resources/_logback-spring.xml [hunk 1]
```diff
@@ -120,11 +112,4 @@
         <resetJUL>true</resetJUL>
     </contextListener>
 
-    <root level="#logback.loglevel#">
```
Comment (deepu105) [inline generators/server/templates/src/main/resources/_logback-spring.xml]: @PierreBesson @jdubois may be the blcok should not be deleted, is CONSOLE the default appender ref for root? Coz when I did this on one of my app where I use a custom console appender like below it doesnt log anything
```
<appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
            <encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
                <layout class="com.**.StackdriverLogbackLayout">
                    <pattern>${FILE_LOG_PATTERN}</pattern>
                    <serviceName>${SERVICE_NAME}</serviceName>
                    <serviceVersion>${SERVICE_VERSION}</serviceVersion>
                </layout>
            </encoder>
        </appender>
```

If I add 
```
    <root>
        <appender-ref ref="CONSOLE"/>
    </root>
```
it works

Comment (jdubois) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: Yes it's supposed to be the default, following the 12 Factors... But I haven't checked in a while. Now that's configured in the application-*.yml files, did you check them?

**Comment by kaidohallik** [comment]
@deepu105 Maybe you removed this line:
```
    <include resource="org/springframework/boot/logging/logback/base.xml"/>
```
Link to this file: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot/src/main/resources/org/springframework/boot/logging/logback/base.xml - this includes `<root>` section.

FYI: in my case I removed this reference to `base.xml` because I wanted custom `file-appender` and now I must manually add this `<root>` section to `logback-spring.xml` file.

File: generators/server/templates/src/main/resources/_logback-spring.xml (hunk 1 above)
Comment (deepu105) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: @jdubois @kaidohallik Also I realized that when setting these in an environment variable to overide it doesn't work
I tried the all of below and they had no effect. I'll probably create an issue in spring boot
```
LOGGING_LEVEL_COM_MYCOMP=INFO
_JAVA_OPTIONS=-Dlogging.level.com.mycomp=INFO
SPRING_APPLICATION_JSON={ "logging.level.com.mycomp": "INFO" }
```

Comment (PierreBesson) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: @deepu105 It doesn't work this way. Those loggers configs are stored in a spring boot `configuration properties class` as maps of which "com.mycomp.*" is a key. And the way you are accessing it, you are requesting the following property : `logging.level.com.mycomp` which doesn't exist.
I checked and you can set it this way:
```
LOGGING_LEVEL_com.netflix.discovery=WARN
_JAVA_OPTIONS=-Dlogging.level."com.netflix.discovery"=WARN
// didn't make it work for the JSON
```
I agree that it's not straightforward. We should write this in our docs.

Comment (jdubois) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: yes +1 for documenting this!

Comment (deepu105) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: @PierreBesson I tried both and doesnt work, first one doesnt normally work due to the . in the name, I can do a workaround but dont want to rely on that as I need to do this in my production environments. For the second one its still picking up without the quotes
```
export _JAVA_OPTIONS=-Dlogging.level."com.mycomp"=WARN && ../gradlew
Picked up _JAVA_OPTIONS: -Dlogging.level.com.mycomp=WARN
```

Comment (deepu105) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: Oh wait I didnt read it properly let me try
`com.mycomp.something`

Comment (deepu105) [reply generators/server/templates/src/main/resources/_logback-spring.xml]: Ok it indeed works after third level :) I wish these where documented properly by Spring

**Comment by GuyJackson** [comment]
Am I right is saying, that if we use the commented out (example prod config) custom appender (in logback-spring.xml):

`

    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
            <fileNamePattern>logFile.%d{yyyy-MM-dd}.log</fileNamePattern>
            <maxHistory>90</maxHistory>
        </rollingPolicy>
        <encoder>
            <charset>utf-8</charset>
            <Pattern>%d %-5level [%thread] %logger{0}: %msg%n</Pattern>
        </encoder>
    </appender>

    <appender name="ASYNC" class="ch.qos.logback.classic.AsyncAppender">
        <queueSize>512</queueSize>
        <appender-ref ref="FILE"/>
    </appender>` 

That the root tag needs to be re-introduced. Something like this:

`<root level="${logging.level.root}">
        <appender-ref ref="ASYNC"/>
    </root>`

**Comment by PierreBesson** [comment]
@GuyJackson if that's the case then it's a bug. Please open a new issue.

**Pull request merged**